gradient/cli/common.py

```diff
@@ -134,20 +152,47 @@ def generate_options_template(ctx, param, value):
         return value
 
     params = {}
+    dataset_value_lists = {}  # Value lists by dataset object key
     for param in ctx.command.params:
         option_name = get_option_name(param.opts)
         if option_name in (OPTIONS_FILE_OPTION_NAME, OPTIONS_DUMP_FILE_OPTION_NAME):
             continue
 
         option_value = ctx.params.get(param.name) or param.default
 
+        # Add any value lists for dataset options to a separate dict
+        paramVars = vars(param)
+        if option_name.startswith("dataset"):
+            # Drop the "dataset" prefix from the *option name*
+            # and use the rest as the dataset object key.
+            object_key = get_object_key("dataset", option_name)
+            dataset_value_lists[object_key] = option_value
+            continue
+
         if isinstance(param.type, cli_types.ChoiceType):
             for key, val in param.type.type_map.items():
                 if val == option_value:
                     option_value = key
 
         params[option_name] = option_value
 
+    # Transform dataset value lists into dataset objects
+    if "uri" in dataset_value_lists:
+        # For each URI, add another dataset object to the list
+        # using values across all option value lists
+        datasets = []
+        for i in range(len(dataset_value_lists["uri"])):
+            dataset = {}
+            for object_key, value_list in dataset_value_lists.items():
+                if i < len(value_list):
+                    dataset[object_key] = value_list[i]
+
+            # Add dataset object to list
+            datasets.append(dataset)
+
+        # Add dataset object array to params
+        params["datasets"] = datasets
+
```

Comment:
It looks fine but could you create a separate class for that? One that inherits from GradientOption. We could then use the new one for those dataset options. I want to be sure it won't make any new code behave in a weird way, like when we add new option that starts with "dataset" somewhere else

Reply:
Think I got it in this next commit. Now if we do another set of options that turn into an object list, they're parameterized.